### PR TITLE
Fixes #2947 find if crc-tray process is running and kill it during cleanup

### LINKED ISSUE #2947: `crc cleanup` command does not remove the tray/app on macOS
Testing the dev tech preview version checking the `crc cleanup` command it was observed that it is not removing the tray on macos anymore. Previous versions for crc remove it so this feature should be paired or re defined.

### PR DESCRIPTION
**Pull request by anjannath**
### How to test
```
# start the tray
$ crc cleanup
# tray should be stopped and no error shown when tray is not running
```

### CODE REVIEW

**Review by gbraad — APPROVED** [review]

**Comment by praveenkumar** [comment]
@anjannath can you fix https://app.circleci.com/pipelines/github/anjannath/crc/827/workflows/47293d69-a498-4d41-a00b-07434785c882/jobs/1821 and also rebase it.

**Comment by praveenkumar** [comment]
@anjannath Doesn't some of the part in this PR depend on #3009 ?

**Review by cfergeau — COMMENTED** [review]
Oops, had this pending for a few days :(

**Comment by anjannath** [comment]
> @anjannath Doesn't some of the part in this PR depend on #3009 ?

yes, #3009 will remove these, we can actually merge just that one

File: pkg/crc/preflight/preflight_checks_tray_darwin.go [hunk 1]
```diff
@@ -119,3 +121,23 @@ func checkIfDaemonRunning() error {
...
+	unLoadTrayAgent()
+
+	processes, err := ps.Processes()
+	if err != nil {
+		fmt.Errorf("unable to find if 'crc-tray' process is running. %w", err)
+	}
+
+	for _, p := range processes {
+		if p.Executable() == constants.TrayBinaryName {
+			_, _, err = crcos.RunWithDefaultLocale("kill", "-9", fmt.Sprintf("%d", p.Pid()))
```
[the remark this answers is not in the document]
Comment (anjannath) [reply pkg/crc/preflight/preflight_checks_tray_darwin.go:135]: the new part is trying to kill tray process that are started by users manually after installation, or processes started again after quiting the instance that was started by launchd at login

`unLoadTrayAgent()` will only kill the ones started by launch

**Comment by anjannath** [comment]
closing as #3009 is going to remove this part of the code anyway

**Pull request closed without merging**